services/api/service.go

```diff
@@ -1209,26 +1375,34 @@ func (api *RelayAPI) handleInternalBuilderStatus(w http.ResponseWriter, req *htt
 		return
 	} else if req.Method == http.MethodPost || req.Method == http.MethodPut || req.Method == http.MethodPatch {
 		args := req.URL.Query()
+		isOptimistic := args.Get("optimistic") == "true"
 		isHighPrio := args.Get("high_prio") == "true"
 		isBlacklisted := args.Get("blacklisted") == "true"
 		api.log.WithFields(logrus.Fields{
 			"builderPubkey": builderPubkey,
+			"isOptimistic":  isOptimistic,
 			"isHighPrio":    isHighPrio,
 			"isBlacklisted": isBlacklisted,
 		}).Info("updating builder status")
-
-		newStatus := datastore.MakeBlockBuilderStatus(isHighPrio, isBlacklisted)
-		err := api.redis.SetBlockBuilderStatus(builderPubkey, newStatus)
+		status := common.LowPrio
+		if isBlacklisted {
+			status = common.Blacklisted
+		} else if isOptimistic {
+			status = common.Optimistic
+		} else if isHighPrio {
+			status = common.HighPrio
+		}
+		err := api.redis.SetBlockBuilderStatus(builderPubkey, status)
 		if err != nil {
-			api.log.WithError(err).Error("could not set block builder status in redis")
+			api.log.WithError(err).Error("could not set builder status in redis")
```

Comment:
Suggest readding the word "block" to reduce the diff.

Reply:
done!